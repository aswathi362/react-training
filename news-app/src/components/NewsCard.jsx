import React, { useEffect } from 'react'

function NewsCard({newsArticle}) {
  return (
    <div className='p-4 bg-white shadow-md rounded-md border border-rose-300'>
        <img src={newsArticle.urlToImage} className='h-1/2 w-full' alt='image not found'/>
        <h2 className='text-lg font-bold'>{newsArticle.title}</h2>
        <p>{newsArticle.description}</p>
        <a 
            href={newsArticle.url} 
            target='_blank' 
            rel='noopener noreferrer' 
            className='text-blue-500'>
                Read More ...
            </a>
    </div>
  )
}

export default NewsCard