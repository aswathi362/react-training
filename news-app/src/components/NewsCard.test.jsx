import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import NewsCard from '../components/NewsCard';

describe('NewsCard Component', () => {
  const mockArticle = {
    title: 'Test News Title',
    description: 'Test News Description',
    urlToImage: 'https://example.com/image.jpg',
    url: 'https://example.com/news',
  };

  test('renders news article information correctly', () => {
    render(<NewsCard newsArticle={mockArticle} />);

    expect(screen.getByText('Test News Title')).toBeInTheDocument();

    expect(screen.getByText('Test News Description')).toBeInTheDocument();

    const readMoreLink = screen.getByText('Read More ...');
    expect(readMoreLink).toBeInTheDocument();
    expect(readMoreLink).toHaveAttribute('href', 'https://example.com/news');

    const image = screen.getByAltText('image not found');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', 'https://example.com/image.jpg');
  });

  test('displays placeholder text for missing image', () => {
    const articleWithoutImage = { ...mockArticle, urlToImage: null };
    render(<NewsCard newsArticle={articleWithoutImage} />);

    expect(screen.getByAltText('image not found')).toBeInTheDocument();
  });

  test('snapshot matches the rendered component', () => {
    const { container } = render(<NewsCard newsArticle={mockArticle} />);
    
    expect(container).toMatchSnapshot();
  });
});
