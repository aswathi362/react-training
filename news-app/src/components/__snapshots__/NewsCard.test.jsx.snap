// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard Component snapshot matches the rendered component 1`] = `
<div>
  <div
    class="p-4 bg-white shadow-md rounded-md border border-rose-300"
  >
    <img
      alt="image not found"
      class="h-1/2 w-full"
      src="https://example.com/image.jpg"
    />
    <h2
      class="text-lg font-bold"
    >
      Test News Title
    </h2>
    <p>
      Test News Description
    </p>
    <a
      class="text-blue-500"
      href="https://example.com/news"
      rel="noopener noreferrer"
      target="_blank"
    >
      Read More ...
    </a>
  </div>
</div>
`;
